source/PlayerInfo.cpp

```diff
@@ -260,7 +260,7 @@ void PlayerInfo::Load(const string &path)
 		else if(child.Token(0) == "ship")
 		{
 			// Ships owned by the player have various special characteristics:
-			ships.push_back(make_shared<Ship>(child));
+			ships.push_back(make_shared<Ship>(child, true));
```

Comment:
I'm curious, what does the true do here? I imagine it's some parameter but idk which

Reply:
So, turns out the Ship loading code assumed that any ship with a variant name was a GameData model ship - you know, from the data directories.  Since variant descriptions are often not complete but instead just state the differences from the base model, the loading code would toss in whatever appeared to be missing.  Like outfits, if none were given.

However, since we now have to track the variant model name for the player's ships (so they can be sold and rebought and everything), we have to put the variant name in the save file.  And we don't want it treated the same - in this case, it's a complete ship definition.  So, since Ship doesn't have a way to tell what file it's loading data from, we pass it a bool to say if the ship definition came from data files (false) or from the player's save file (true).

Reply:
Oh right this calls the constructor, ofc
I assume you added a comment explaining this somewhere?

Reply:
I left it to the variable name:

`Ship(const DataNode &node, bool isSavedShip = false);`